File: Opsi.Pocos/Resource.cs
namespace Opsi.Pocos;

public class Resource
{
    public virtual string FullName { get; set; } = default!;

    public string? AssignedBy { get; set; }

    public DateTime? AssignedOnUtc { get; set; }

    public string? AssignedTo { get; set; }

    public string? CreatedBy { get; set; } = default!;

    public Guid ProjectId { get; set; }

    public List<ResourceVersion> ResourceVersions { get; set; }

    public Resource()
    {
        ResourceVersions = new List<ResourceVersion>();
    }
}

File: Opsi.Web/src/lib/Models/Resource.ts
import type ResourceVersion from "./ResourceVersion";

export default class Resource {
  assignedBy?: string;
  assignedOnUtc?: Date;
  assignedTo?: string;
  fullName?: string;
  projectId?: string;
  resourceVersions: ResourceVersion[]
  username?: string;
  versionId?: string;
  versionIndex?: number;

  constructor() {
    this.assignedBy = undefined;
    this.assignedOnUtc = undefined;
    this.assignedTo = undefined;
    this.fullName = undefined;
    this.projectId = undefined;
    this.resourceVersions = [];
    this.username = undefined;
    this.versionId = undefined;
    this.versionIndex = undefined;
  }
}